File: ch7/7-6.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const MaxLine = 100

func main() {
	if len(os.Args) != 3 {
		fmt.Printf("error: the program takes 2 arguments\n")
		os.Exit(1)
	}
	f1, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Printf("%s\n", err)
		os.Exit(1)
	}
	f2, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Printf("%s\n", err)
		os.Exit(1)
	}
	defer func() {
		var err error
		err = f1.Close()
		if err != nil {
			fmt.Printf("%s\n", err)
			os.Exit(1)
		}
		err = f2.Close()
		if err != nil {
			fmt.Printf("%s\n", err)
			os.Exit(1)
		}
	}()
	s1 := bufio.NewScanner(f1)
	s2 := bufio.NewScanner(f2)
	for s1.Scan() && s2.Scan() {
		l1 := s1.Text()
		l2 := s2.Text()
		if l1 == l2 {
			fmt.Printf("%s\n", l1)
			fmt.Printf("%s\n", l2)
			return
		}
	}
	if err = s1.Err(); err != nil {
		fmt.Printf("%s\n", err)
		os.Exit(1)
	}
	if err = s2.Err(); err != nil {
		fmt.Printf("%s\n", err)
		os.Exit(1)
	}
	fmt.Printf("Identical lines\n")
}

File: kr/ch7/7-6.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAXLINE 1024

int main(int argc, char *argv[])
{
    FILE *f1, *f2;
    char line1[MAXLINE], line2[MAXLINE];
    char *prog = argv[0];

    if (argc != 3) {
        printf("error: the program takes 2 arguments\n");
        return 0;
    }

    if ((f1 = fopen(*++argv, "r")) != NULL) {
        if ((f2 = fopen(*++argv, "r")) != NULL) {
            while (fgets(line1, MAXLINE, f1) != NULL && fgets(line2, MAXLINE, f2) != NULL) {
                if (strcmp(line1, line2) != 0) {
                    printf("%s", line1);
                    printf("%s", line2);

                    fclose(f1);
                    fclose(f2);

                    return 0;
                }
            }
        } else {
            fprintf(stderr, "%s can't open %s", prog, *argv);
            exit(1);
        }
    } else {
        fprintf(stderr, "%s can't open %s", prog, *argv);
        exit(1);
    }

    printf("Identical lines\n");

    return 0;
}
